Update SoC branch with new changes in main

arinagarwal opened the pull request:
Branch is currently outdated, so I want to pull from main

Chase-Arline committed:
Main FW: small clean-up changes

Mostly style changes / unused variables removed. One of the can
decoding functions was using pointers to the buffer instead of the dereferenced values.

Chase-Arline committed:
Main FW: Added basic FreeRTOS, untested currently

Basic import of a Teensy4 FreeRTOS library
Modified tasks to use FreeRTOS
Cleaned up unnecessary files for the display

Chase-Arline committed:
Main FW: improve RTOS tasks, Precharge logic

The precharge firmware was not very clean. It still needs some work done
but should be easier to follow now.
Made the RTOS tasks enter loops so that they will execute properly.

Chase-Arline committed:
Main FW: Define RTOS task stack sizes, basic testing

The display task + datalogging should have a larger stack size as they
need underlying tft and sd card data buffers. Some basic testing was
done without being connected to the larger system.

Chase-Arline committed:
RTOS: add SPI mutex to avoid bus conflicts

Bus conflicts were occuring in SPI between the display and datalog task.
The mutex now provides safe access to the bus.

Chase-Arline committed:
Datalog: Save files every 1 second

Before we were assuming that the task would for sure run 20 times a
second. This is not 100% guaranteed due to the new mutex, so we should
save based on the clock instead.

Chase-Arline committed:
teensy RTC not working, board is asking for sync from pc every reset instead of using the RTC from VBAT

Commit:
merged display changes and added aux battery changes that Trevor made too

Commit:
quick pin fix in display.h

Commit:
one last time?

Commit:
Sorted all display changes to 4.1, Chase feel free to add all comments as needed

Commit:
fast display working, don't see any crashing at the moment

Commit:
Pushing latest display integration. This should have proper erasure and have display fully optimized for debugging. Next step is to integrate the speedometer for display and then display is done.

Commit:
Added speedometer, added bunch of TODOs to implement angle kill switch

Commit:
Partial Documentation, but finalized race-ready code

Commit:
Documentation 1/3 done. Styling code to properly use the header files fully done.

Commit:
Frankly, I'm too tired of doing documentation. Please someone else finish this. I'm gonna sleep

Commit:
Everything documented except Can.ino and PreCharge.ino

Commit:
No Can.ino or Precharge.ino documentation

Commit:
Added another display todo. Sorry lol

Commit:
Added Doxygen configuration file into Documentation folder, use this to use Doxygen

Commit:
Testing something

Commit:
Github tutorial

Commit:
first commit

caslam committed:
Merge pull request #5 from Washington-Superbike/github_tutorial

first commit

caslam committed:
Display changes

* Removed enum and some extra variables

* Shortened thermistor printing method a bit

* Slapped config.h back in
Change screen here with #define

* Update for speedometer screen
Added the main battery voltage and made RPM a bit smaller to make room
for other info. Speed doesn't reprint until it changes.

Chase-Arline committed:
Repo: remove unnecessary 'PCBS' directory

The PCBS directory is very outdated since we moved to Altium

Chase-Arline committed:
repo: remove unnecessary documetation files

Vast majority of our documentation can be done via normal commenting,
the exceptions can be added in via other means (HTML pages, etc.)

Chase-Arline committed:
Restructure Part 1

The FW has gotten bloated, and needs restructuring.

First, by using a shared context variable, we can reduce the amount of
shared pointers to variables.

Second, we should not be defining any variables in header files, only
declarations and preprocessor directives.

Third, we should start removing unnecessary or unclear comments.
The code should be straight forward enough to read that comments are
largely not needed, except for complex sections.

Fourth, we should be using TickType_t to time events instead of
millis().

Chase-Arline committed:
Merge pull request #10 from Washington-Superbike/user/carline/remove-pcbs-dir

Repo: remove unnecessary 'PCBS' directory

Chase-Arline committed:
Merge pull request #9 from Washington-Superbike/user/carline/remove-documentation

repo: remove unnecessary documetation files

Chase-Arline commented:
Can do a reset instead 
i.e. 
`git checkout main` 
`git pull`
`git checkout SoC` 
`git reset --hard main`
`git checkout -b user/arinagarwal/reset_soc`
 `git push`
 
 Otherwise SoC will include additional commits from whoever was last working on it

arinagarwal commented:
Done, thank you for the guidance!

On Sun, May 21, 2023 at 3:11 PM Chase Arline ***@***.***>
wrote:

> Can do a reset instead
> i.e.
> git checkout main
> git pull
> git checkout SoC
> git reset --hard main
> git checkout -b user/arinagarwal/reset_soc
> git push
>
> Otherwise SoC will include additional commits from whoever was last
> working on it
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Washington-Superbike/ARHMA2021/pull/16#issuecomment-1556306807>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJFJBKVIEPASA3FYM6O7GHDXHKHK3ANCNFSM6AAAAAAYJUNZDU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Chase-Arline commented:
Cool, once you have something you want to merge into main, just switch this PR from `main --> SoC` to `user/arinagarwal/reset_sec -> main`

arinagarwal commented:
Sounds good

On Tue, May 23, 2023 at 12:21 AM Chase Arline ***@***.***>
wrote:

> Cool, once you have something you want to merge into main, just switch
> this PR from main --> SoC to user/arinagarwal/reset_soc -> main
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Washington-Superbike/ARHMA2021/pull/16#issuecomment-1558682659>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJFJBKXERA5VBUBEQJNTZEDXHRQP3ANCNFSM6AAAAAAYJUNZDU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

caslam committed:
using the other FreeRTOS library

caslam committed:
a few functionality changes

requests/reads battery voltages from BMS, gyro angle check moved
to error state, and stays in error state until the switch is turned off

caslam committed:
Merge pull request #8 from Washington-Superbike/user/carline/restructure

Restructure Part 1
